src/main/java/info/mywinecellar/config/WebSecurityConfig.java

```diff
@@ -47,8 +50,15 @@ SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
                 .logoutSuccessUrl("/")
                 .permitAll();
 
+        // Logged user only access
+        http.authorizeRequests().antMatchers("/bottle/**", "/tasted/**", "/wishlist/**")
+                .hasAnyRole("USER", "ADMIN");
+
         // Admin only access
         http.authorizeRequests().antMatchers("/admin/**").hasRole("ADMIN");
+        http.authorizeRequests().antMatchers("/country/**").hasRole("ADMIN");
```

Comment:
You should look at the Thymleaf files to see what we're going for here.  But these endpoints below are meant to be accessed by any authenticated or non-authenticated person, not just ADMIN.

Reply:
Right now, as a non-authenticated user is possible to edit any of these entities, which doesn't seem right. 
For example:  
- http://localhost:8080/country/1/edit 
- http://localhost:8080/grape/1/edit 
- http://localhost:8080/area/1/edit 

On the other hand, the only `/country` endpoints are meant just for editing, meanwhile, non-authenticated or authenticated users can still access through `/d` resources.
The same goes for the area and grapes as specified above.

Reply:
@pauldpearson did you have the chance to check my last comments?

Reply:
@matiaspompilio Hi yes and sorry for delay, quite busy all around!  ADMIN user is a hyper specific and limited role in this app that really at this point isn't even fully implemented or thought out yet.  

For the rest I think what would be best is to limit access based on the request mappings for authenticated (role=USER) or non-authenticated 'users'.  

http.authorizeRequests().antMatchers(HttpMethod.POST, "/country/*/edit").hasRole("USER");

Hope that helps.

Reply:
Hi @pauldpearson thanks for your reply. I added new changes based on your last suggestion.
Edit access is limited only to users as you commented. Please take a look.
Thank you!